service/history/transferQueueTaskExecutorBase.go

```diff
@@ -282,3 +301,29 @@ func (t *transferQueueTaskExecutorBase) deleteExecution(
 		forceDeleteFromOpenVisibility,
 	)
 }
+
+func (t *transferQueueTaskExecutorBase) isCloseExecutionTaskPending(ms workflow.MutableState, weCtx workflow.Context) (bool, error) {
+	closeTransferTaskId := ms.GetExecutionInfo().CloseTransferTaskId
+	// taskID == 0 if workflow still running in passive cluster or closed before this field was added (v1.17).
+	if closeTransferTaskId == 0 {
+		return false, nil
+	}
+	currentClusterName := t.shard.GetClusterMetadata().GetCurrentClusterName()
+	nsActive := ms.GetNamespaceEntry().ActiveInCluster(currentClusterName)
+	// In passive cluster, transfer task queue check can be ignored but not visibility task queue.
+	if !nsActive {
+		return false, nil
+	}
```

Comment:
If the check only applies to active task/ns, then I think it should live in activeTaskExecutor instead of executorBase which is shared by both active and standby.

We can also apply the check to both active and standby I guess? To me, it seems like even for standby, close execution task should be executed before deletion. I don't remember why for DeleteWorkflowExecution API, we only check for active. cc @alexshtin @yux0

Reply:
I think we should check this in standby. But need to align with Alex on this.

Reply:
Will wait to hear back from Alex

Reply:
Yes, those needs to be splitted. Also visibility queue checks should be in visibility queue, not here.

Reply:
Ok, I moved the whole check into just the active task executor, and I no longer query ActiveInCluster.